zerver/tests/test_thumbnail.py

```diff
@@ -88,12 +88,12 @@ def test_thumbnail_redirect_for_spectator(self) -> None:
         self.assertEqual(response.status_code, 403)
         remove_ratelimit_rule(86400, 0, domain="spectator_attachment_access_by_file")
 
-        # Deny random file access
+        # redirect random file access to imageNotFound.png
         response = self.client_get(
             "/thumbnail",
             {
                 "url": "user_uploads/2/71/QYB7LA-ULMYEad-QfLMxmI2e/zulip-non-existent.txt",
                 "size": "full",
             },
         )
-        self.assertEqual(response.status_code, 403)
+        self.assertEqual(response.status_code, 302)
```

Comment:
We don't want to serve a 302 redirect to a 200.  We want clients to _know_ that the request had a permission denied, and they cannot know that if they see s 302 -> 200.  It's very important that we be honest and accurate with our response codes.

Our goal is to serve a response with a status of 403, a content-type of either `text/plain` or `image/png`, and a content which matches that content-type.